gateway-service-impl/src/main/java/org/hypertrace/gateway/service/entity/query/ExecutionTreeBuilder.java

```diff
@@ -78,12 +78,13 @@ public QueryNode build() {
     // the data store
     if (singleSourceForAllAttributes.isPresent()) {
       String source = singleSourceForAllAttributes.get();
-      QueryNode selectionAndFilterNode = buildExecutionTreeForSameSourceFilterAndSelection(source);
+      QueryNode rootNode = buildExecutionTreeForSameSourceFilterAndSelection(source);
       if (LOG.isDebugEnabled()) {
-        LOG.debug("Execution Tree:{}", selectionAndFilterNode.acceptVisitor(new PrintVisitor()));
+        LOG.debug("Execution Tree:{}", rootNode.acceptVisitor(new PrintVisitor()));
       }
 
-      return selectionAndFilterNode;
+      rootNode.acceptVisitor(new ExecutionContextBuilderVisitor(executionContext));
+      return buildExecutionTree(executionContext, rootNode);
```

Comment:
This is much cleaner now. Can you just add a comment on top of the `filterTree.acceptVisitor(new OptimizingVisitor())` at line 100. Mention that this is needed to merge filters corresponding to the same source into one DataFetcherNode instead of multiple for each filter.